### PR TITLE
feat: Literal support

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/processing/op_code.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+#include "op_code.h"
+
+#include <limits>
+
+#include "context/instruction.h"
+
+namespace hlasm_plugin::parser_library::processing {
+
+unsigned char processing_status_cache_key::generate_reladdr_bitmask(context::id_index id)
```
Comment (michalbali256) [inline parser_library/src/processing/op_code.cpp:23]: I think this function could use a comment explaining what it does

File: parser_library/test/processing/mach_instr_test.cpp [hunk 2]
```diff
@@ -232,3 +232,9 @@ SYM  DS   CL1
     a.collect_diags();
     ASSERT_EQ(a.diags().size(), 0U);
 }
+
+TEST(mach_instr_processing, verify_rel_addr_size)
+{
+    for (const auto& instr : context::instruction::machine_instructions)
```
Comment (michalbali256) [inline parser_library/test/processing/mach_instr_test.cpp:238]: are we `EXPECT`ing anything here? or just relying on the assert in `generate_reladdr_bitmask`?

File: parser_library/src/context/literal_pool.cpp [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            ord_ctx.align(lit_val.align_on_halfword ? halfword : no_align),
+            symbol_attributes(symbol_origin::DAT,
+                ebcdic_encoding::a2e[(unsigned char)lit->get_type_attribute()],
+                lit->get_length_attribute(solver)),
+            {});
+
+        ord_ctx.reserve_storage_area(size, no_align);
+
+        if (!cycle_ok)
+            diags.add_diagnostic(diagnostic_op::error_E033(it->second.r));
```
Comment (michalbali256) [inline parser_library/src/context/literal_pool.cpp:155]: test this case?

File: parser_library/src/expressions/mach_expr_term.cpp [hunk 4]
```diff
@@ -160,44 +208,161 @@ mach_expr_data_attr::mach_expr_data_attr(
...
+        auto& dd = lit->get_data_definition();
+        context::symbol_attributes attrs(context::symbol_origin::DAT,
+            ebcdic_encoding::a2e[(unsigned char)dd.get_type_attribute()],
+            dd.get_length_attribute(solver),
+            dd.get_scale_attribute(solver),
+            dd.get_integer_attribute(solver));
+        if ((attribute == context::data_attr_kind::S || attribute == context::data_attr_kind::I)
+            && !attrs.can_have_SI_attr())
+        {
+            add_diagnostic(diagnostic_op::warning_W011(get_range()));
```
Comment (michalbali256) [inline parser_library/src/expressions/mach_expr_term.cpp:278]: test this diagnostic?

File: parser_library/src/expressions/mach_expr_term.cpp [hunk 5]
```diff
@@ -99,40 +135,46 @@ context::dependency_collector mach_expr_self_def::get_dependencies(context::depe
...
         return *location_counter;
 }
 
-void mach_expr_location_counter::fill_location_counter(context::address addr) { location_counter = std::move(addr); }
-
 const mach_expression* mach_expr_location_counter::leftmost_term() const { return this; }
 
 void mach_expr_location_counter::apply(mach_expr_visitor& visitor) const { visitor.visit(*this); }
 
+size_t mach_expr_location_counter::hash() const { return (size_t)0x0009459ca772d69b; }
```
[the remark this answers is not in the document]
Comment (slavek-kucera) [reply parser_library/src/expressions/mach_expr_term.cpp:174]: These are literally random numbers, it does not matter what they are exactly, they should be just different (ideally even after the size_t cast on 32-bit platforms).

File: parser_library/src/context/literal_pool.cpp (hunk 3 above)
Comment (slavek-kucera) [reply parser_library/src/context/literal_pool.cpp:155]: I am reasonably sure that this should never really happen, but not 100%.

**Review by michalbali256 — APPROVED** [review]

**Review by michalbali256 — APPROVED** [review]

**Pull request merged**